name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/wizards/datatransfer/PDFImportAssistant.java

```diff
@@ -0,0 +1,156 @@
+                return new CommerzbankPDFExctractor(client);
+            case "consorsbank": //$NON-NLS-1$
+                return new ConsorsbankPDFExctractor(client);
+            case "dab": //$NON-NLS-1$
+                return new DABPDFExctractor(client);
+            case "degiro": //$NON-NLS-1$
+                return new DegiroPDFExtractor(client);
+            case "db": //$NON-NLS-1$
+                return new DeutscheBankPDFExctractor(client);
+            case "dkb": //$NON-NLS-1$
+                return new DkbPDFExtractor(client);
+            case "fintechgroupbank": //$NON-NLS-1$
+                return new FinTechGroupBankPDFExtractor(client);
+            case "ingdiba": //$NON-NLS-1$
+                return new INGDiBaExtractor(client);
+            case "onvista": //$NON-NLS-1$
+                return new OnvistaPDFExtractor(client);
+            case "sbroker": //$NON-NLS-1$
+                return new SBrokerPDFExtractor(client);
+            case "unicredit": //$NON-NLS-1$
+                return new UnicreditPDFExtractor(client);
+            default:
+                return new AssistantPDFExtractor(client);
+        }
+    }
+
+    public static String DetectBankIdentifier(File PDFfilename) throws IOException  
+    {  
+
+        // Loading an existing document   
+        PDDocument document = PDDocument.load(PDFfilename);  
+
+        // Getting the PDDocumentInformation object  
+        PDDocumentInformation pdd = document.getDocumentInformation();  
+
+        // Getting the PDFTextStripper object  
+        PDFTextStripper textStripper = new PDFTextStripper();  
+        textStripper.setSortByPosition(true);  
+        String text = textStripper.getText(document);  
+        document.close();  
+
+        // PDF import assistant - Level 1 - Allocate by PDF Author  
+        String PDFauthor;  
+        if (pdd.getAuthor() == null) PDFauthor = "";  
+        else PDFauthor = pdd.getAuthor();  
+
+        switch (PDFauthor)  
+        {  
+
+            case "Scalable Capital Vermögensverwaltung GmbH": //$NON-NLS-1$  
+                return "baaderbank";  
+            case "Consorsbank": //$NON-NLS-1$  
+                return "consorsbank";  
+            case "Computershare Communication Services GmbH": //$NON-NLS-1$  
+                return "dab";  
+            case "DKB AG": //$NON-NLS-1$  
+                return "dkb";  
+            case "ING-DiBa": //$NON-NLS-1$  
+                return "ingdiba";  
+
+            default:  
+
+                // PDF import assistant - Precheck if specific securities could cause faulty detection  
+                // ISIN DE0005088108 = Baader Bank Aktie detect the bank identifier "Baader Bank"   
+                Matcher matcherISIN = Pattern.compile("DE0005088108|DE0005428007|DE000CBK1001|FR0000131104|INE007B01023|DE000FTG1111|CH0001351862|CH0001350328|CH0001354296|CH0001352720|CH0001350112|CH0318681860|CH0001343885|FR0000131104|INE007B01023").matcher( text );  
```

Comment:
Diese Zeile verstehe ich noch nicht komplett. Du meinst wenn man in einem Depot der Commerzbank die "Baader Bank" Aktie kauft, dann könnte es falsch erkannt werden?

Reply:
Damit versuche ich Sonderfälle abzufangen. Bspw. wenn das Dokument von der "ING DiBa" kommt und das Wertpapier mit der ISIN DE0005088108 und der Beschreibung "Baader Bank Aktie" enthalten sein könnte, dann wäre das erste Suchergebnis das der "Baader Bank" anstelle der "ING DiBa".

Da das PDF dann im falschen Extractor landet und nicht lesbar wäre, soll der Anwender besser die Auswahl übernehmen.